### PR TITLE
Fix headers in PrestoRequest

### PR DESCRIPTION
**Pull request by ggreg**
Ensure headers are always passed in HTTP GET and POST requests.

### CODE REVIEW

File: prestodb/client.py [hunk 1]
```diff
@@ -273,12 +271,12 @@ def process(self, http_response):
...
-        if 'X-Presto-Clear-Session' in http_response.headers:
-            propname = response.headers['X-Presto-Clear-Session']
+        if constants.HEADER_CLEAR_SESSION in http_response.headers:
+            propname = response.headers[constants.HEADER_CLEAR_SESSION]
             self._session_properties.pop(propname, None)
 
-        if 'X-Presto-Set-Session' in http_response.headers:
-            set_session_header = response.headers['X-Presto-Set-Session']
+        if constants.HEADER_SET_SESSION in http_response.headers:
+            set_session_header = response.headers[constants.HEADER_SET_SESSION]
```
Comment (haozhun) [inline prestodb/client.py]: Do you need to update `self._headers[constants.HEADER_SESSION]` here?

File: prestodb/client.py [hunk 2]
```diff
@@ -273,12 +271,12 @@ def process(self, http_response):
         if 'error' in response:
             raise self._process_error(response['error'])
 
-        if 'X-Presto-Clear-Session' in http_response.headers:
-            propname = response.headers['X-Presto-Clear-Session']
+        if constants.HEADER_CLEAR_SESSION in http_response.headers:
+            propname = response.headers[constants.HEADER_CLEAR_SESSION]
             self._session_properties.pop(propname, None)
```
Comment (haozhun) [inline prestodb/client.py]: Do you need to update `self._headers[constants.HEADER_SESSION]` here?

**Comment by haozhun** [comment]
ping @ggreg

File: prestodb/client.py (hunk 2 above)
Comment (ggreg) [inline prestodb/client.py]: I am adding a `ClientSession` class and referencing it as `PrestoRequest._client_session`. `PrestoRequest.headers` is then a Python property attribute built from the latest state of the `ClientSession` instance.

**Comment by ggreg** [comment]
Opening a new pull request from a branch.

**Pull request closed without merging**